client/components/purchase-detail/index.jsx

```diff
@@ -31,20 +35,32 @@ const PurchaseDetail = ( {
 
 	if ( buttonText || isPlaceholder ) {
 		buttonElement = (
-			<Button
-				className="purchase-detail__button"
+			<PurchaseButton
 				disabled={ isSubmitting }
 				href={ href }
 				onClick={ onClick }
 				target={ target }
-				primary>
-				{ buttonText }
-			</Button>
+				text={ buttonText } />
 		);
 	}
 
+	const renderBody = () => {
```

Comment:
I think there's enough rendering logic here, to split this out to render() and renderBody().